### ISSUE TITLE
False positive SC2094

### ISSUE THREAD

**Issue by hgl** [koalaman/shellcheck#2604]
#### For bugs
- Rule Id: SC2094
- My shellcheck version: online 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

while read -r args; do
  set -- $args
  for f; do
    [ -d "$f" ]
  done
done <"$f"

```

#### Here's what shellcheck currently says:

```
Line 4:
    [ -d "$f" ]
         ^-- [SC2094](https://www.shellcheck.net/wiki/SC2094) (info): Make sure not to read and write the same file in the same pipeline.
 
Line 6:
done <"$f"
      ^-- [SC2094](https://www.shellcheck.net/wiki/SC2094) (info): Make sure not to read and write the same file in the same pipeline.
```

#### Here's what I wanted or expected to see:

No warning, since there is really no writing to f.

**Comment by trothwell** [comment]
Is the form of the sample the best way to handle this pattern? Are there alternatives?

**Comment by hgl** [comment]
The "form" boils down to reusing a one-time redirected variable, which I personally incline to favor if the variable is only temporary.

This sample can be simplified and generates the same false positive:

```
#!/bin/sh

while read -r f; do
  "$f"
done <"$f"
```

**Comment by martinetd** [comment]
( This is a dup of #1379 )
